tensorflow/python/ops/parallel_for/control_flow_ops.py

```diff
@@ -104,9 +105,18 @@ def while_body(i, *ta_list):
   assert len(output) in (0, len(flat_loop_fn_dtypes))
   if not output:
     # This may happen for the case where iters == 0.
-    return None
-  else:
-    return nest.pack_sequence_as(loop_fn_dtypes, output)
+    # Pack a list of empty tensors with the proper ranks to match pfor output on 0 iters
+    loop_var = array_ops.placeholder_with_default(0, shape=[])
+    try:
+      loop_fn_out = loop_fn(loop_var)
+    except InvalidArgumentError: 
+      loop_fn_out = array_ops.placeholder_with_default(0, shape=[])
+    out_shapes = [[0]+ops.convert_to_tensor(x).shape
+            for x in nest.flatten(loop_fn_out)]
+    output = [array_ops.zeros(out_shapes[i], dt)
+            for i,dt in enumerate(flat_loop_fn_dtypes)]
+
+  return nest.pack_sequence_as(loop_fn_dtypes, output)
```

Comment:
Thanks for the fix, and we are running the checks to see if there are other tests that also need attention.

Also am wondering
- Could we catch more/all exceptions in addition to `InvalidArgumentError` as there is not clear contract about which exception type should be thrown for  `loop_fn` when the input is empty ?

- Is it possible that the, in the exception case,  line 116-117 would fail for `len(flat_loop_fn_dtypes) != out_shapes` as we computed `out_shapes` based on the `loop_fn_out` of empty shape placeholder.   In this case, could we directly use a logic to compute the output as an empty tensor with no shape in the `except` block ?

Reply:
1. Yes that makes sense, changed to catch all Exceptions.
2. I'm not sure, but directly computing the output appears clearer anyway, so I have made that change as well